sources/niLang/src/FilePattern.cpp

```diff
@@ -502,8 +502,8 @@ class cFilePatternRegex : public cIUnknownImpl<ni::iRegex> {
       const achar* p = _vPatterns[i].Chars();
       if (*p == '!') {
         ++p;
-        if (afilepattern_match(p,aaszString)) {
-          return eFalse;
+        if (!afilepattern_match(p,aaszString)) {
+          return eTrue;
```

Comment:
I'm not sure I understand what case this covers. Can we add a Test_Regex.cpp or Test_FilePattern.cpp in niLang/tsrc that shows the issue this resolves? Should be fairly quick, you can copy/paste Test_FileMemory.cpp as a starting point (that's a simple/short fixture).